README.md

```diff
@@ -2,3 +2,163 @@
 
 An initial scalable filesystem for Linux operating systems
 
+- [Why use initoverlayfs to generate initramfs images?](#why-use-initoverlayfs-to-generate-initramfs-images-)
+- [Installation](#installation)
+    * [Step 1 - Deploy the software](#step-1---deploy-the-software)
+    * [Step 2 - Run initoverlayfs-install](#step-2---run-initoverlayfs-install)
+    * [Step 3 - Reboot to test](#step-3---reboot-to-test)
+    * [Step 4 - Validating the boot](#step-4---validating-the-boot)
+
+# Why use initoverlayfs to generate initramfs images?
+
+An initramfs (Initial RAM File System) image is a fundamental component in preparing Linux systems during the boot process, preceding the initiation of the init process. 
+
+Typically, generating an initramfs involves assembling all available kernel modules and necessary files to boot and support any hardware using the specific Linux kernel version XYZ.  
+
+However, this conventional approach presents a significant challenge: loading such a voluminous image into memory during boot is time-consuming and can be problematic in critical scenarios such as healthcare, automotive, and aviation.
+
+Conversely, the initoverlayfs approach proposes a solution: "**dividing the initramfs image into two parts**." 
+
+This division entails segregating the initramfs image into two distinct components.
+The first component houses only the kernel modules necessary for storage, responsible for bringing up the operating system. Subsequently, it triggers the second component, containing all additional kernel modules and essential files required to support the Linux boot process. 
+
+This innovative approach serves to diminish the size of the initramfs image, thus enhancing the speed of the boot process.
+
+For illustration, consider a comparison of the sizes using dracut versus initoverlayfs:
+
+**Using Dracut**:
+``` bash
+# dracut -f
+# du -sh /boot/initramfs-6.5.5-300.fc39.x86_64.img
+36M	/boot/initramfs-6.5.5-300.fc39.x86_64.img
+```
+
+**Using initoverlayfs**:
+``` bash
+# /usr/bin/initoverlayfs-install
+# du -sh /boot/initramfs-6.5.5-300.fc39.x86_64.img
+13M	/boot/initramfs-6.5.5-300.fc39.x86_64.img 
```

Comment:
@masneyb is looking into things to make this even smaller. Eventually we will likely be able to remove systemd also, leaving all systemd stuff in initoverlayfs. Note the goal is not to rewrite systemd, we just need some things to spin up storage quickly and then switch to systemd (or alternative init process, none of this solution is systemd specific).

Eventually I expect this to be single-digit M, like < 10M.

Reply:
This will be very nice!